apps/theming/lib/Jobs/MigrateBackgroundImages.php

```diff
@@ -27,56 +27,96 @@
 namespace OCA\Theming\Jobs;
 
 use OCA\Theming\AppInfo\Application;
-use OCP\App\IAppManager;
 use OCP\AppFramework\Utility\ITimeFactory;
 use OCP\BackgroundJob\IJobList;
 use OCP\BackgroundJob\QueuedJob;
 use OCP\Files\AppData\IAppDataFactory;
+use OCP\Files\IAppData;
 use OCP\Files\NotFoundException;
 use OCP\Files\NotPermittedException;
 use OCP\Files\SimpleFS\ISimpleFolder;
-use OCP\IConfig;
+use OCP\IDBConnection;
+use Psr\Log\LoggerInterface;
 
 class MigrateBackgroundImages extends QueuedJob {
 	public const TIME_SENSITIVE = 0;
 
-	private IConfig $config;
-	private IAppManager $appManager;
+	protected const STAGE_PREPARE = 'prepare';
+	protected const STAGE_EXECUTE = 'execute';
+	// will be saved in appdata/theming/global/
+	protected const STATE_FILE_NAME = '25_dashboard_to_theming_migration_users.txt';
```

Comment:
```suggestion
	protected const STATE_FILE_NAME = '25_dashboard_to_theming_migration_users.json';
```
As it contains JSON data?